### PR TITLE
[Mesheryctl] View/Delete Configurations 

### LINKED ISSUE #5928: [mesheryctl] improvements in ```mesheryctl``` sub commands
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior

For ```pattern``` command - 
1) User can't view pattern on basis of pattern-id 
image

2) User can't delete pattern on basis of [patternName/patternId]
image

For ```filter``` command , user can't view and delete filter on basis of [patternId/patternName]

more details TBA.

#### Desired Behavior

```pattern``` & ```filter``` sub commands should perfectly work for the above use case. 

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- 📺 [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- ⌨️ [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- ⌨️ [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by harkiratsm**
Signed-off-by: harkiratsm <multaniharry714@gmail.com>

**Description**

This PR fixes #5928 

**Notes for Reviewers**

- View, Delete the configurations on the based of **id** of particular configurations.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by alphaX86** [comment]
Lint checks are failing, just conveying in advance :)

**Comment by harkiratsm** [comment]
@alphaX86 still some work pending until i hand over is to for the review .

**Comment by leecalcote** [comment]
Labeling as `draft`, until @harkiratsm signals for review. In the meantime, I'll suggest that @alphaX86 or @acald-creator might want to collaborate on this branch/PR.

**Comment by warunicorn19** [comment]
@harkiratsm would you please bring this topic up in the Meshery dev call today?

**Comment by harkiratsm** [comment]
@warunicorn19 Already taken this topic on previous dev call.

File: mesheryctl/internal/cli/root/filter/delete.go [hunk 1]
```diff
@@ -27,17 +29,46 @@ mesheryctl exp filter delete test-wasm
...
+			}
+		}
+
+		// Delete the filter using the id
+		if isID {
+			err := utils.DeleteConfiguration(filter, "filter")
+			if err != nil {
+				return errors.Wrap(err, utils.SystemError(fmt.Sprintf("failed to delete filter %s", args[0])))
+			}
+			fmt.Printf("Filter %s deleted successfully\n", args[0])
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/filter/delete.go:62]: We don’t use printf, do we?

@alphaX86 @nishantwrp @hexxdump

Comment (alphaX86) [reply mesheryctl/internal/cli/root/filter/delete.go:62]: Seems in some areas, it's been left unchecked...

Comment (leecalcote) [reply mesheryctl/internal/cli/root/filter/delete.go:62]: @alphaX86 will you offer a bit of guidance to align with mesheryctl coding conventions? 

Correcting this in any other areas that have slipped by makes for good Hacktoberfest issues.

Comment (alphaX86) [reply mesheryctl/internal/cli/root/filter/delete.go:62]: Oh sure!

Comment (harkiratsm) [reply mesheryctl/internal/cli/root/filter/delete.go:62]: @leecalcote that's my mistake of using printf, would make the changes accordingly.

**Comment by Chadha93** [comment]
@harkiratsm Let's discuss this PR in today's dev call.

**Comment by harkiratsm** [comment]
@leecalcote I have checked the ```perf``` behaviour is amazing, but what I am doing here is some what identical to the other ```ctl``` for eg docker cli does . 

Deleting on basis of the **id**.

**Comment by leecalcote** [comment]
@harkiratsm great! Now, start doing what the rest of the mesheryctl does. :wink:

**Comment by harkiratsm** [comment]
- for viewing with **--all**  I will change the behaviour to somewhat around perf .
- For deleting, let's leave it onto to like other **cli** does because it's more viable.

**Comment by harkiratsm** [comment]
For the above  **first** point I would track that in a separately PR #6328.

**Comment by leecalcote** [comment]
@alphaX86 Thumbs-up?

File: mesheryctl/pkg/utils/helpers.go [hunk 2]
```diff
@@ -446,6 +447,65 @@ func StringInSlice(str string, slice []string) bool {
 	return false
 }
 
+// GetID returns a array of IDs from meshery server endpoint /api/{configurations}
+func GetID(configuration string) ([]string, error) {
+	url := MesheryEndpoint + "/api/" + configuration + "?page_size=10000"
+	configType := configuration + "s"
```
Comment (hershd23) [inline mesheryctl/pkg/utils/helpers.go:453]: Although I have no major issues with it, hardcoding an `s` at the end seems rather hacky, would it be feasible to change something in the payload we get?

If you think what I suggested is unnecessary you can ignore it.
If you think that it makes sense but isn't urgent, you can add a TODO comment and create an issue ticket at an appropriate time
Else you can make this change in this PR

**Comment by hershd23** [comment]
Reviewing it @leecalcote

**Review by hershd23 — COMMENTED** [review]
One comment rest LGTM

File: mesheryctl/pkg/utils/helpers.go (hunk 2 above)
Comment (alphaX86) [reply mesheryctl/pkg/utils/helpers.go:453]: +1 for this, although it's a minor one, we might need to have a look

**Review by alphaX86 — COMMENTED** [review]
Rest looks fine to me

File: mesheryctl/pkg/utils/helpers.go (hunk 2 above)
Comment (harkiratsm) [reply mesheryctl/pkg/utils/helpers.go:453]: @hershd23 @alphaX86 The reason to use ```s``` was because of the response we are getting.

**Comment by harkiratsm** [comment]
@leecalcote We are good to go :+1:

**Review by alphaX86 — APPROVED** [review]

File: mesheryctl/internal/cli/root/app/offboard.go [hunk 3]
```diff
@@ -40,6 +42,36 @@ mesheryctl app offboard -f [filepath]
 			return errors.Wrap(err, "error processing config")
 		}
 
+		app := ""
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/app/offboard.go:45]: I see this particular code block getting repeated multiple times, can this be abstracted into a single function?
utils.ValidId(configtype string) (string, error)?

**Review by alphaX86 — COMMENTED** [review]
Changes look good. But I sense a lint error caught in the checks... could you investigate it @harkiratsm ? Only after the holidays of course ;)

**Pull request merged**